h2o-core/src/main/java/hex/ModelBuilder.java

```diff
@@ -504,29 +509,31 @@ public static void bulkBuildModels(String modelType, Job job, ModelBuilder<?, ?,
     }
     for( int i=0; i<N; ++i ) //all sub-models must be completed before the main model can be built
       try {
-        submodel_tasks[i].join();
+        final H2O.H2OCountedCompleter task = submodel_tasks[i];
```

Comment:
How can the task be `null`?

Reply:
not anymore, because I'm now throwing an exception instead of breaking the previous loop, but anyway, I prefer to keep the safe-check as it is so easy to reintroduce this bug.

Reply:
do we enable asserts in prod? I'm fine with asserts instead

Reply:
asserts are only enabled in tests (in R h2o is also started with asserts enabled)

Reply:
Will the assert be added?

Reply:
it's been added below

Reply:
closed

Reply:
closed

Reply:
Great!